anyway/widgets/suburban_widgets/killed_and_injured_count_per_age_group_widget.py

```diff
@@ -0,0 +1,201 @@
+from collections import defaultdict
+from datetime import datetime
+from typing import Callable, Dict, Tuple, List
+
+from flask_babel import _
+from flask_sqlalchemy import BaseQuery
+from sqlalchemy import func, asc
+
+from anyway.app_and_db import db
+from anyway.backend_constants import BE_CONST, InjurySeverity as IS
+from anyway.models import InvolvedMarkerView
+from anyway.request_params import RequestParams
+from anyway.utilities import parse_age_from_range
+from anyway.widgets.suburban_widgets.sub_urban_widget import SubUrbanWidget
+from anyway.widgets.widget import register
+from anyway.widgets.widget_utils import (
+    order_by_list_stack_bar_widget,
+    gen_entity_labels,
+    add_empty_keys_to_gen_two_level_dict,
+)
+
+# RequestParams is not hashable, so we can't use functools.lru_cache
+cache_dict = {}
+AGE_RANGE_DICT = {0: 4, 5: 14, 15: 19, 20: 24, 25: 44, 45: 64, 65: 200}
+
+
+@register
+class KilledInjuredCountPerAgeGroupStackedWidget(SubUrbanWidget):
+    name: str = "killed_and_injured_count_per_age_group_stacked"
+
+    def __init__(self, request_params: RequestParams):
+        super().__init__(request_params, type(self).name)
+        self.rank = 30
+
+    def generate_items(self) -> None:
+        raw_data = filter_and_group_injured_count_per_age_group(self.request_params)
+
+        partial_processed = add_empty_keys_to_gen_two_level_dict(
+            raw_data,
+            get_age_range_list(),
+            IS.codes(),
+        )
+
+        structured_data_list = []
+        for age_group, severity_dict in partial_processed.items():
+            series_severity = [
+                {"label_key": IS(severity_value).get_label(), "value": count}
+                for severity_value, count in severity_dict.items()
+            ]
+            structured_data_list.append({"label_key": age_group, "series": series_severity})
+
+        self.items = order_by_list_stack_bar_widget(
+            structured_data_list,
+            [
+                IS.KILLED.get_label(),
+                IS.SEVERE_INJURED.get_label(),
+                IS.LIGHT_INJURED.get_label(),
+            ],
+        )
+
+    @staticmethod
+    def localize_items(request_params: RequestParams, items: Dict) -> Dict:
+        items["data"]["text"] = {
+            "title": _("Killed and injury stacked per age group in ")
+            + request_params.location_info["road_segment_name"],
+            "labels_map": gen_entity_labels(IS),
+        }
+        return items
+
+
+@register
+class KilledInjuredCountPerAgeGroupWidget(SubUrbanWidget):
```

Comment:
Our new standard is single class per file

Reply:
Those 2 widgets share the same code (all the code that is outside of the class's), that why i created then in the same file.
I can separate them, what do you think?

Reply:
Looking at the widget classes themselves, their code is very similar, with small differences. (I mean, many of the future change will need to be done twice, in both). Maybe it better to have single code, with one or two parameters, that does it. And then have two classes that call it and provide these parameters. That code can be in
1 - a different class that both inherit, or
2 - a third class that both use as a member, or
3 - a file of utilities that both classes use. 
I would prefer the second option